### PR TITLE
Fixes #510: Paginate My(milestone, campaign, donation, community) pages 

### LINKED ISSUE #510: Make Manage pages paginated
**Task:**
Make the My Donations, My Delegations, My Campaigns and My Communities paginated the same way as the My Milestone page is.

**Work Estimate:**
2-4h

**Original description:**
https://etherscan.io/tx/0x66bc674ddb2d2c142462c0eb2ad4ed1f5d99cd4fec87c23d7dba72b8bc53fb86

Not in my delegations... likely related to: https://github.com/Giveth/giveth-dapp/issues/467

### PR DESCRIPTION
**Pull request by anshumanv**
Fixes #510

### CODE REVIEW

**Comment by anshumanv** [comment]
@vojtechsimetka updated the branch. :+1:

**Comment by vojtechsimetka** [comment]
Hey @anshumanv, thanks for sorting this out! I just discovered one bug (potentially). To test the functionality, I just modified the amount of items per page (which should actually be configurable from props I think) https://github.com/Giveth/giveth-dapp/pull/528/files#diff-dafd855049382453b5da27d6c216042fR39 .

The whole system bugged completely. As you can see in the video

https://www.dropbox.com/s/ohmaznupdt9ujej/Screen%20Recording%202018-11-16%20at%2014.10.11.mov?dl=0

**Comment by anshumanv** [comment]
Ah :confused: silly mistake

Thanks for pointing to that! Fixed :sparkles:

**Comment by GriffGreen** [comment]
how do we get this netlified :-D so excited about being ablt to move all my donations around!

**Comment by vojtechsimetka** [comment]
Working on it now @GriffGreen

File: src/components/views/MyCampaigns.jsx [hunk 1]
```diff
@@ -234,12 +235,12 @@ class MyCampaigns extends Component {
                           </tbody>
                         </table>
 
-                        {campaigns.data.length > campaigns.itemsPerPage && (
+                        {campaigns.total > campaigns.limit && (
```
Comment (vojtechsimetka) [inline src/components/views/MyCampaigns.jsx]: Should be campaigns.total > this.state.itemsPerPage

File: src/components/views/MyCampaigns.jsx [hunk 2]
```diff
@@ -234,12 +235,12 @@ class MyCampaigns extends Component {
...
                           <center>
                             <Pagination
-                              activePage={campaigns.skipPages + 1}
-                              itemsCountPerPage={campaigns.itemsPerPage}
-                              totalItemsCount={campaigns.totalResults}
+                              activePage={campaigns.skip + 1}
```
Comment (vojtechsimetka) [inline src/components/views/MyCampaigns.jsx]: Needs to be divided by campaigns.limit . Also it is very strange to mix the service counting the records and the component counting records.

File: src/components/views/MyDACs.jsx [hunk 3]
```diff
@@ -163,12 +164,12 @@ class MyDACs extends Component {
...
                         </table>
 
-                        {dacs.data.length > dacs.itemsPerPage && (
+                        {dacs.total > dacs.limit && (
                           <center>
                             <Pagination
-                              activePage={dacs.skipPages + 1}
-                              itemsCountPerPage={dacs.itemsPerPage}
-                              totalItemsCount={dacs.totalResults}
+                              activePage={dacs.skip + 1}
```
Comment (vojtechsimetka) [inline src/components/views/MyDACs.jsx]: Needs to be divided by dacs/limit . Similar comment as to the campaigns part.

**Comment by vojtechsimetka** [comment]
Merged in 48f2441cc602d0e36070ae18779d2790040f1b36

**Comment by anshumanv** [comment]
Thanks for the follow-up @vojtechsimetka, I'll be more cautious next time. Great to see it make it to the release. :tada: :smile:

**Pull request closed without merging**